inference-engine/ie_bridges/rust/crates/openvino-sys/build.rs

```diff
@@ -0,0 +1,258 @@
+use std::path::{Path, PathBuf};
+
+use bindgen;
+use cmake;
+
+fn main() {
+    // Trigger rebuild on changes to build.rs and Cargo.toml and every source file.
+    println!("cargo:rerun-if-changed=build.rs");
+    println!("cargo:rerun-if-changed=Cargo.toml");
+    let cb = |p: PathBuf| println!("cargo:rerun-if-changed={}", p.display());
+    visit_dirs(Path::new("src"), &cb).expect("to visit source files");
+
+    // Generate bindings from C header.
+    let openvino_c_api_header =
+        file("upstream/inference-engine/ie_bridges/c/include/c_api/ie_c_api.h");
+    let bindings = bindgen::Builder::default()
+        .header(openvino_c_api_header.to_string_lossy())
+        // While understanding the warnings in https://docs.rs/bindgen/0.36.0/bindgen/struct.Builder.html#method.rustified_enum
+        // that these enums could result in unspecified behavior if constructed from an invalid
+        // value, the expectation here is that OpenVINO only returns valid layout and precision
+        // values. This assumption is reasonable because otherwise OpenVINO itself would be broken.
+        .rustified_enum("layout_e")
+        .rustified_enum("precision_e")
+        .rustified_enum("resize_alg_e")
+        .parse_callbacks(Box::new(bindgen::CargoCallbacks))
+        .generate()
+        .expect("generate C API bindings");
+    let out = PathBuf::from(std::env::var("OUT_DIR").unwrap());
+    bindings
+        .write_to_file(out.join("bindings.rs"))
+        .expect("failed to write bindings.rs");
+
+    // Find OpenVINO libraries, either pre-installed or by building from source.
+    let library_paths = if let Some(path) = std::env::var_os("OPENVINO_INSTALL_DIR") {
+        // Given a path to an OpenVINO installation (e.g. https://docs.openvinotoolkit.org/latest/openvino_docs_install_guides_installing_openvino_apt.html),
+        // find the paths to the necessary libraries and add them to the library search path.
+        // It would be preferable to use pkg-config here to retrieve the libraries when they are
+        // installed system-wide but there are issues:
+        //  - OpenVINO does not install itself a system library, e.g., through ldconfig.
+        //  - OpenVINO relies on a `plugins.xml` file for finding target-specific libraries
+        //    and it is unclear how we would discover this in a system-install scenario.
+        let deployment_tools = dir(path).join("deployment_tools");
+        let openvino_libraries = deployment_tools.join("inference_engine/lib/intel64");
+        let tbb_libraries = deployment_tools.join("inference_engine/external/tbb/lib");
+        let ngraph_libraries = deployment_tools.join("ngraph/lib");
+        vec![openvino_libraries, tbb_libraries, ngraph_libraries]
+    } else {
+        // Build OpenVINO with CMake.
```

Comment:
If you cd to a subdir of `OUT_DIR` and then use `cmake /path/to/upstream && make` the build dir should be in that subdir of `OUT_DIR` and nothing outside `OUT_DIR` should be modified, which means that `--no-verify` is not necessary when publishing the crate.

Reply:
I think that's effectively what `let mut config = cmake::Config::new("upstream");` below. The build artifacts are all contained within `OUT_DIR` so that is fine; what isn't as fine is that OpenVINO actually downloads things into its own tree, which is what Cargo then complains about. I opened https://github.com/openvinotoolkit/openvino/issues/2316 to address this and once I get a reply I can just direct CMake to download that stuff to some directory in `OUT_DIR`.